### DISCUSSION TITLE
about visualDL or visualize loss curve(如何可視化loss曲線?)

### DISCUSSION THREAD

**Discussion by t109598032** [PaddlePaddle/PaddleOCR#14238] [Q&A] [answered]
無論yml參數檔案設定``Global.use_visualdl``為True或False
開始訓練時皆會顯示: ``ppocr WARNING: You are using VisualDL, the VisualDL is deprecated and removed in ppocr!``
請問目前ppocr無法使用visualDL來可視化訓練Loss曲線了嗎?
有什麼其他方法可以查看training loss curve以便決定是否可以停止訓練?

**Comment by Liyulingyue** [comment]
根据数据记录相关的代码，PPOCR应该除了 VDL之外，还有wandb和loggers可以看。
```
from .vdl_logger import VDLLogger
from .wandb_logger import WandbLogger
from .loggers import Loggers
```

**Reply by star20245** [reply, marked as answer]
源码中有你可以看下，把以下配置加入就可以，wandb建议离线部署，在执行训练之前需要登录wandb，关于wandb得相关可以自行查询资料
Global:
  ... ...
  use_wandb: True #启用wandb
  ... ...
wandb:
    project: paddle  #在wandb中创建的项目名称
    entity: XXXX#账号名
    name: a1 #数据展示标签栏
    dir: ./output/pgnet_r50_vd_totaltext/wandb_logs #可以不配置,不配置默认会存放在“{save_model_dir}/wandb”路径下
    resume: auto
